lib/src/over_react_redux/over_react_redux.dart

```diff
@@ -0,0 +1,393 @@
+  ReduxProviderProps([Map props]) : this.props = props ?? new JsBackedMap();
+
+  @override
+  final Map props;
+
+  @override
+  ReactComponentFactoryProxy get componentFactory => ReactJsReactReduxComponentFactoryProxy(JsReactRedux.Provider, isProvider: true);
+
+  @override
+  String get propKeyNamespace => '';
+
+  Store get store => props['store'];
+  set store(Store v) => props['store'] = v;
+
+  dynamic get context => props['context'];
+  set context(dynamic v) => props['context'] = v;
+}
+
+UiFactory<ReduxProviderProps> ReduxProvider = ([Map map]) => new ReduxProviderProps(map);
+
+class ReactJsReactReduxComponentFactoryProxy extends ReactJsContextComponentFactoryProxy {
+  /// The JS class used by this factory.
+  @override
+  final ReactClass type;
+
+  @override
+  final Function factory;
+
+  @override
+  final bool shouldConvertDomProps;
+
+  @override
+  final bool isConsumer;
+
+  @override
+  final bool isProvider;
+
+  ReactJsReactReduxComponentFactoryProxy(
+    ReactClass jsClass, {
+    this.shouldConvertDomProps: true,
+    this.isConsumer: false,
+    this.isProvider: false,
+  }) : this.type = jsClass,
+        this.factory = React.createFactory(jsClass),
+        super(jsClass, isProvider: isProvider, isConsumer: isConsumer, shouldConvertDomProps: shouldConvertDomProps);
+
+  @override
+  ReactElement build(Map props, [List childrenArgs]) {
+    return super.build(_generateReduxJsProps(props), childrenArgs);
+  }
+
+  /// Returns a JavaScript version of the specified [props], preprocessed for consumption by ReactJS and prepared for
+  /// consumption by the [react] library internals.
+  Map _generateReduxJsProps(Map props) {
+    JsBackedMap propsForJs = new JsBackedMap.from(props);
+
+    if (propsForJs['store'] != null) {
+      propsForJs['store'] = _reduxifyStore(propsForJs['store']);
+    }
+
+    if (propsForJs['context'] != null) {
+      propsForJs['context'] = propsForJs['context'].jsThis;
+    }
+
+    return propsForJs;
+  }
+}
+
+_reduxifyStore(Store store){
+  return JsReactReduxStore(
+    getState: allowInterop(() {
+      return wrapInteropValue(store.state);
+    }),
+    subscribe: allowInterop((cb) {
+        return allowInterop(store.onChange.listen((_){cb();}).cancel);
+    }),
+    dispatch: allowInterop((action) {
+      store.dispatch(action);
+    })
+  );
+}
+
+
+@JS()
+@anonymous
+class JsReactReduxStore {
+  external factory JsReactReduxStore({
+    ReactInteropValue Function() getState,
+    void Function(dynamic) dispatch,
+    void Function(Function) subscribe,
+  });
+}
+
+dynamic _convertArgsToChildren(List childrenArgs) {
+  if (childrenArgs.isEmpty) {
+    return null;
+  } else if (childrenArgs.length == 1) {
+    return childrenArgs.single;
+  } else {
+    markChildrenValidated(childrenArgs);
+    return childrenArgs;
+  }
+}
+
+@visibleForTesting
+@JS()
+@anonymous
+class JsConnectOptions {
+    bool Function(ReactInteropValue, ReactInteropValue) areStatesEqual;
```

Comment:
These need to be `external` and getters/setters to work.
```suggestion
    external get bool Function(ReactInteropValue, ReactInteropValue) areStatesEqual;
    external set areStatesEqual(bool Function(ReactInteropValue, ReactInteropValue) value);
```

Are these even used, though? If not, they could be removed

Reply:
Yes, the variable names are optional in the new syntax